libpromises/conversion.c

```diff
@@ -407,15 +406,14 @@ long IntFromString(const char *s)
         return (long) CFSTARTTIME;
     }
 
-    remainder[0] = '\0';
 
-    sscanf(s, "%lld%c%s", &a, &c, remainder);
+    sscanf(s, "%lld%c%c", &a, &c, &remainder);
 
 // Test whether remainder is space only
 
-    if ((a == CF_NOINT) || (!IsSpace(remainder)))
+    if ((a == CF_NOINT) || (remainder == ' '))
```

Comment:
Logic error, you want exactly the opposite. And don't compare with space, but do `!isspace(remainder)`. See `ctype.h` for more such functions.